rmw_fastrtps_shared_cpp/src/listener_thread.cpp

```diff
@@ -100,13 +100,14 @@ rmw_fastrtps_shared_cpp::join_listener_thread(rmw_context_t * context)
   return RMW_RET_OK;
 }
 
-#define TERMINATE(msg) \
-  do { \
+#define TERMINATE_THREAD(msg) \
+  { \
     RCUTILS_SAFE_FWRITE_TO_STDERR( \
       RCUTILS_STRINGIFY(__FILE__) ":" RCUTILS_STRINGIFY(__function__) ":" \
-      RCUTILS_STRINGIFY(__LINE__) RCUTILS_STRINGIFY(msg) ": terminating ..."); \
-    std::terminate(); \
-  } while (0)
+      RCUTILS_STRINGIFY(__LINE__) RCUTILS_STRINGIFY(msg) \
+      ": ros discovery info listener thread will shutdown ...\n"); \
+    break; \
```

Comment:
@ivanpauno meta: this is certainly better than aborting, but I'm not thrilled with the idea of relying on a log to be seen for the user to do something about it i.e. client code will never know this happened.

Reply:
I think that's the trade-of, either you finish abruptly or you log and continue.
The main reason I prefer logging and continue, is that everything else can actually continue working correctly without this thread running. Some graph introspection methods will not return correct information, but everything else will continue working correctly.